ansq/utils.py

```diff
@@ -179,9 +180,14 @@ def get_logger(
     return logger
 
 
-def truncate_text(text: str, limit: int = 256) -> str:
+def truncate_text(
+    text: typing.Union[bytes, str], limit: int = 256
+) -> typing.Union[bytes, str]:
     """Truncate a given `text` if the `limit` is reached"""
     if limit <= 0:
         raise ValueError("limit must be greater than 0")
 
-    return text[:limit] + "..." if len(text) > limit else text
+    if len(text) <= limit:
+        return text
+
+    return text[:limit] + b"..." if isinstance(text, bytes) else text[:limit] + "..."
```

Comment:
```suggestion
    return text[:limit] + (b"..." if isinstance(text, bytes) else "...")
```

Reply:
BTW, it looks like we can safely let go the `str` type here. I'd like to see the library operating binary data only.

Reply:
>  I'd like to see the library operating binary data only.

Good suggestion! Addressed in b36b63d